_static/js/custom.js

```diff
@@ -237,4 +237,13 @@ $(document).ready(() => {
   document.head.appendChild(instantPageScript);
+ 
+  let currentState = history.state;
+  // Remove domain and 'highlight' tag from URL.
+  let newURL = window.location.href.replace("https://docs.godotengine.org", "");
+  const reg = new RegExp('[?]highlight=([^?]*)');
+  newURL = newURL.replace(reg, "");
+
+  // Replace URL without reloading page.
+  history.replaceState(currentState, "", newURL);
```

Comment:
Sphinx has a built-in `hideSearchWords()` helper function (seen [here](https://github.com/sphinx-doc/sphinx/blob/1660b3ebc64b12f5ab2146ef9e37adf01904922d/sphinx/themes/basic/static/doctools.js#L162)) that could be used to remove the highlights and trim the URL automatically!

```suggestion
  const url = new URL(location.href);
  if (url.searchParams.has('highlight')) {
    Documentation.hideSearchWords();
  }
```